pkgs/development/python-modules/tensorflow/default.nix

```diff
@@ -1,187 +1,112 @@
-
-    bazelTarget = "//tensorflow/tools/pip_package:build_pip_package";
-
-    meta = with stdenv.lib; {
-      description = "Computation using data flow graphs for scalable machine learning";
-      homepage = "http://tensorflow.org";
-      license = licenses.asl20;
-      maintainers = with maintainers; [ jyp abbradar ];
-      platforms = with platforms; if cudaSupport then linux else linux ++ darwin;
-    };
-  };
-
-in buildPythonPackage (common // {
+buildPythonPackage rec {
   pname = "tensorflow";
-  version = common.version;
-  name = "tensorflow-${common.version}";
-
-  deps = stdenv.mkDerivation (common // {
-    name = "tensorflow-external-${common.version}";
-
-    nativeBuildInputs = common.nativeBuildInputs ++ [ bazel ];
+  version = "1.4.1";
+  name = "${pname}-${version}";
+  format = "wheel";
+  disabled = ! (isPy35 || isPy36 || isPy27);
 
-    preConfigure = common.preConfigure + ''
-      export PYTHON_LIB_PATH="$(pwd)/site-packages"
-    '';
-
-    buildPhase = ''
-      mkdir site-packages
-      bazel --output_base="$(pwd)/output" fetch $bazelFlags $bazelTarget
-    '';
-
-    installPhase = ''
-      rm -rf output/external/{bazel_tools,\@bazel_tools.marker,local_*,\@local_*}
-      # Patching markers to make them deterministic
-      for i in output/external/\@*.marker; do
-        sed -i 's, -\?[0-9][0-9]*$, 1,' "$i"
-      done
-      # Patching symlinks to remove build directory reference
-      find output/external -type l | while read symlink; do
-        ln -sf $(readlink "$symlink" | sed "s,$NIX_BUILD_TOP,NIX_BUILD_TOP,") "$symlink"
-      done
-
-      cp -r output/external $out
-    '';
-
-    dontFixup = true;
-
-    outputHashMode = "recursive";
-    outputHashAlgo = "sha256";
-    outputHash = "0xs2n061gnpizfcnhs5jjpfk2av634j1l2l17zhy10bbmrwn3vrp";
-  });
-
-  nativeBuildInputs = common.nativeBuildInputs ++ [ (bazel.override { enableNixHacks = true; }) ];
-
-  configurePhase = ''
-    runHook preConfigure
-    export PYTHON_LIB_PATH="$out/${python.sitePackages}"
-    ./configure
-    runHook postConfigure
-  '';
-
-  buildPhase = ''
-    mkdir -p output/external
-    cp -r $deps/* output/external
-    chmod -R +w output
-    find output -type l | while read symlink; do
-      ln -sf $(readlink "$symlink" | sed "s,NIX_BUILD_TOP,$NIX_BUILD_TOP,") "$symlink"
-    done
-
-    patchShebangs .
-    find -type f -name CROSSTOOL\* -exec sed -i \
-      -e 's,/usr/bin/ar,${binutils}/bin/ar,g' \
-      {} \;
-
-    mkdir -p $out/${python.sitePackages}
-    bazel --output_base="$(pwd)/output" build $bazelFlags $bazelTarget
-
-    bazel-bin/tensorflow/tools/pip_package/build_pip_package $PWD/dist
-  '';
-
-  # tensorflow depends on tensorflow_tensorboard, which cannot be
+  # cudatoolkit is split (see https://github.com/NixOS/nixpkgs/commit/bb1c9b027d343f2ce263496582d6b56af8af92e6)
+  # However this means that libcusolver is not loadable by tensor flow. So we undo the split here.
+  cudatoolkit_joined = symlinkJoin {
```

Comment:
This needs to be wrapped into `if cudaSupport` or smth like that

Reply:
Is that really so? I thought that `nix` had lazy evaluation.

Reply:
I think all derivation (or `stdenv.mkDerivation`?) attributes are forced to become environment variables in a generated build script.